### DISCUSSION TITLE
If Auto Complete Not working in Your IDE

### DISCUSSION THREAD

**Discussion by EG-Mohamed** [Laravel-Backpack/community-forum#349] [Tips & Tricks]
If AutoComplete in IDE such as phpStorm etc. ,  Is not working You can use this amazing helper [laravel-ide-helper](https://github.com/barryvdh/laravel-ide-helper)

## Installation

Require this package with composer using the following command:

```bash
composer require --dev barryvdh/laravel-ide-helper
```

### Automatic PHPDoc generation for Laravel Facades

You can now re-generate the docs yourself (for future updates)

```bash
php artisan ide-helper:generate
```

Easy Right? Happy Coding ♥

**Comment by imasubmarine** [comment]
Hello,

Im using Laravel 9 and VSCode.

For some reason, each time i open VSCode i have to do again _php artisan ide-helper:generate_ 
Although it generates exactly the same _ide_helper.php file that the one already there.
If not i have tons of infos like _Non static method should not be called statically_ using Backpack Facades
(intelephense) and no autocompletion.

I only have this problem with Backpack CRUD facade.

Any ideas why ?

**Reply by EG-Mohamed** [reply]
honestly, I'm not used VScode before hope some one else can help you asap

**Reply by imasubmarine** [reply]
@EGYWEB-Mohamed  thanks anyway!

**Reply by pxpm** [reply]
@imasubmarine I use both this packages:
![image](https://user-images.githubusercontent.com/7188159/211525167-86dbd032-b6ca-4e74-beb4-937b55d245e1.png)

With the package @EGYWEB-Mohamed shared. It does not require me to run the command again when opening. 

I think I remember that it happened to me before too, and what I did was to run the command first, and only after let the packages do the indexing. It's have been working until now!

Cheers

**Reply by pxpm** [reply]
@imasubmarine also, have a look at https://marketplace.visualstudio.com/items?itemName=georgykurian.laravel-ide-helper

Basically it runs the command automatically for you on save, so you get the completion anyway

**Reply by imasubmarine** [reply]
thanks @pxpm !
I will give it a try.
